storage/registry.go

```diff
@@ -40,8 +40,18 @@ func InitializeWithSafeTermination(ctx context.Context, s Storage, options *Sett
 	}, wg)
 
 	if err := s.Open(options); err != nil {
-		return fmt.Errorf("error opening storage: %s", err)
+		return nil, fmt.Errorf("error opening storage: %s", err)
 	}
 
-	return nil
+	var decoratedRepository TransactionalRepository
```

Comment:
idea of the decorators was to allow hiding the wrapping of strorages/ repositories so that creating the NewEncryptingRepository does not depend on the storage being open